File: trunk/main.net/src/Coherence.Tools/Core/Extractor/MvelExtractor.cs
using Seovic.Core.Expression;

namespace Seovic.Core.Extractor
{
    /// <summary>
    /// An imlementation of <see cref="IExtractor"/> that extracts value 
    /// from a target object using <see cref="MvelExpression"/>.
    /// </summary>
    public class MvelExtractor : ExpressionExtractor
    {
        /// <summary>
        /// Deserialization constructor (for internal use only).
        /// </summary>
        public MvelExtractor()
        {
        }

        /// <summary>
        /// Construct an <c>MvelExtractor</c> instance.
        /// </summary>
        /// <param name="expression">The expression to evaluate.</param>
        public MvelExtractor(string expression)
            : base(new MvelExpression(expression))
        {
        }
    }
}

File: trunk/main/src/com/seovic/core/extractor/MvelExtractor.java
/*
 * Copyright 2009 Aleksandar Seovic
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.seovic.core.extractor;


import com.seovic.core.Extractor;
import com.seovic.core.expression.MvelExpression;


/**
 * An imlementation of {@link Extractor} that extracts value from a target
 * object using {@link MvelExpression}.
 *
 * @author Aleksandar Seovic  2009.11.07
 */
public class MvelExtractor<T>
        extends ExpressionExtractor<T>
    {
    // ---- constructors ----------------------------------------------------

    /**
     * Deserialization constructor (for internal use only).
     */
    public MvelExtractor()
        {
        }

    /**
     * Construct an <tt>MvelExtractor</tt> instance.
     *
     * @param expression  the expression to use
     */
    public MvelExtractor(String expression)
        {
        super(new MvelExpression<T>(expression));
        }
    }